### PR TITLE
fix make for golang 1.20

### PR DESCRIPTION
**Pull request by dmolenda-sumo**
**Description:** <Describe what has changed.>
fixing a bug for golang 1.20
 math/rand is deprecated and should not be used in golang 1.20. I changed it to crypto/rand and thanks to that make on golag 1.20 works

**Link to tracking Issue:** #20341

**Testing:**  run make

### CODE REVIEW

**Comment by atoulme** [comment]
LGTM. Thank you for the fix - is there any way to avoid panics?

File: exporter/instanaexporter/internal/converter/span_converter_test.go [hunk 1]
```diff
@@ -255,13 +255,17 @@ func TestSpanWithError(t *testing.T) {
 }
 
 func generateTraceID() (data [16]byte) {
-	rand.Read(data[:])
-
+	_, err := rand.Read(data[:])
+	if err != nil {
```
Comment (codeboten) [inline exporter/instanaexporter/internal/converter/span_converter_test.go:259]: might be best to return the error and use `require.NoError` in the caller. what do you think?

File: exporter/azuredataexplorerexporter/adx_exporter_test.go [hunk 2]
```diff
@@ -163,7 +163,8 @@ func TestIngestedDataRecordCount(t *testing.T) {
 		ingestOptions: ingestOptions,
 		logger:        logger,
 	}
-	rand.Seed(time.Now().UTC().UnixNano())
+	source := rand.NewSource(time.Now().UTC().UnixNano())
+	rand := rand.New(source)
```
Comment (Aneurysm9) [inline exporter/azuredataexplorerexporter/adx_exporter_test.go:167]: Again, please don't shadow package names with variable names.

File: exporter/instanaexporter/internal/converter/span_converter_test.go [hunk 3]
```diff
@@ -253,9 +260,10 @@ func TestSpanWithError(t *testing.T) {
 	spanSlice := ptrace.NewSpanSlice()
 
 	sp1 := spanSlice.AppendEmpty()
-	setupSpan(sp1, SpanOptions{
+	err := setupSpan(sp1, SpanOptions{
 		Error: "some error",
 	})
+	require.NoError(t, err)
```
Comment (dmolenda-sumo) [inline exporter/instanaexporter/internal/converter/span_converter_test.go:266]:
```suggestion
	require.NoError(t, setupSpan(sp1, SpanOptions{
		Error: "some error",
	}))
```

**Review by codeboten — APPROVED** [review]

**Comment by dmolenda-sumo** [comment]
Is it necessary for someone else to review this pull request before merging the changes?

**Review by codeboten — APPROVED** [review]

**Pull request merged**